src/container/RegisterCard/CardNumberContainer.tsx

```diff
@@ -0,0 +1,84 @@
+import { useFilter, useForm } from '../../hooks';
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { Input, InputContainer } from '../../components/form';
+import { RegisterCardType } from '../../pages/RegisterCard';
+import { Validator } from '../../domain';
+
+const MAX_LENGTH = 4;
+const VALIDATE_ERROR = '카드 번호를 입력 해 주세요.';
+
+export default function CardNumberContainer({ onChange }: RegisterCardType) {
+  const { formToArray, onlyNumber } = useFilter();
+  const { isEnterCardNumber } = Validator();
+  const [errorMessage, setErrorMessage] = useState(VALIDATE_ERROR);
+  const cardNumber2 = useRef(null);
+  const cardNumber3 = useRef(null);
+  const cardNumber4 = useRef(null);
+  const [cardNumber, setCardNumber] = useForm({
+    first: '',
+    second: '',
+    third: '',
+    fourth: '',
+  });
+
+  const isValidateCard = useMemo(() => (
+    isEnterCardNumber(cardNumber, MAX_LENGTH)
+  ), [cardNumber]);
+  const cardNumbers = useMemo(() => (
+    formToArray(cardNumber).join('')
+  ), [cardNumber]);
+
+  useEffect(() => {
+    onChange(cardNumbers);
+
+    if (!isValidateCard) {
+      setErrorMessage(VALIDATE_ERROR);
+      return;
+    }
+
+    setErrorMessage('');
```

Comment:
이런 구문은... 이렇게도 괜찮아보입니다.

```suggestion
    setErrorMessage(isValidateCard ? '' : VALIDATE_ERROR);
```